src/torznab.ts

```diff
@@ -523,8 +523,14 @@ async function fetchCaps(indexer: {
 }): Promise<Caps> {
 	let response;
 	try {
+		const abortController = new AbortController();
+		setTimeout(
+			() => void abortController.abort(),
+			ms("10 seconds"),
+		).unref();
 		response = await fetch(
 			assembleUrl(indexer.url, indexer.apikey, { t: "caps" }),
+			{ signal: abortController.signal },
```

Comment:
I learned about `AbortSignal.timeout()` recently. It's a lot simpler than the whole AbortController dance. 

```suggestion
			{ signal: AbortSignal.timeout(ms("10 seconds"))},
```